src/utils.rs

```diff
@@ -16,64 +16,77 @@ pub fn secs_to_duration(s: f64) -> Duration {
 }
 
 pub struct Estimate {
-    buf: Vec<f64>,
-    buf_cap: usize,
-    last_idx: usize,
+    buf: Box<[f64; 15]>,
```

Comment:
The mechanisms you're using here could use a bunch of comments -- it's not at all clear to me what the approach is. Also, do you have an actual use case where this matters or do you just like squeezing out performance? :)

Reply:
I do like squeezing out performance and that's the main reason, however there were a few issues related to performance and this is a small stepping stone in improving it.

Reply:
I'm curious what kind of scenarios you're seeing where issues like these impact performance. Can you say more about that?